client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -226,6 +231,10 @@ private synchronized WinRm createService() {
 
         BindingProvider bp = (BindingProvider)winrm;
         bp.getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, endpoint.toExternalForm());
+        if (requestTimeoutInSeconds != null) {
+            //BindingProviderProperties.REQUEST_TIMEOUT
+            bp.getRequestContext().put("com.sun.xml.internal.ws.request.timeout", requestTimeoutInSeconds * 1000);
```

Comment:
We are polling for the result, so this might actually be working, but not having the expected result. It applies to single request only, which returns immediately. You'll have to add additional logic in `WinRmClient` to account for the remaining time for each request.

Might also be worth adding a sleep after a receive response, which I just noticed is missing, using a Repeater with a back off delay.